### ISSUE TITLE
Travis awareness of test changes

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#4951]
Many of our failures in nightly testing are due to minor mistakes in adding new tests or modifying existing tests.  At present, our Travis runs only test a few things, just to make sure things aren't completely broken.  In the event that tests themselves are changed in a given PR, it would be nice if Travis would run those tests or perhaps the directories that contain them.  While this would not provide airtight testing coverage (and realistically, little will), it would help by catching low-hanging fruit that did not show up in the developer's environment for some reason (e.g., developer changed a correctness test but forgot to check the corresponding performance test mode; developer failed to commit a new file that the test depended on; etc.)

**Comment by ben-albrecht** [comment]
Here is a data point for when this Travis feature could have saved us some testing noise: https://github.com/chapel-lang/chapel/pull/15074

**Comment by tzinsky** [comment]
We no longer use Travis for CI, and have shifted to a local hosted version of Jenkins.   Closing this issue as it is no longer relevant to the environment we now operate in.

**Comment by lydia-duncan** [comment]
Tim, do we have a Github Actions specific or more general version of the request itself?

**Comment by bradcray** [comment]
I agree that the original intent of this request is still valid even if the mention of Travis is outdated.  I'd probably just reopen it editing "GitHub actions" or "Chapel's CI" into the place of "Travis"

**Comment by bradcray** [comment]
I've edited the subject and OP to reframe the request without mentioning a specific CI technology.
